File: go/proverb/proverb.go
// Package proverb provides functionality to make proverbs
package proverb

// Proverb makes a proverb with the given rhymes
func Proverb(rhyme []string) []string {
	size := len(rhyme)
	if size < 1 {
		return nil
	}

	var res []string
	for i := 0; i < size-1; i++ {
		res = append(res, "For want of a "+rhyme[i]+" the "+rhyme[i+1]+" was lost.")
	}
	res = append(res, "And all for the want of a "+rhyme[0]+".")
	return res
}

File: exercises/practice/proverb/.meta/src/reference/java/Proverb.java
final class Proverb {
    private String[] words;

    Proverb(String[] words) {
        this.words = words;
    }

    String recite() {
        if (words.length < 1) {
            return "";
        }
        final StringBuilder result = new StringBuilder();
        for (int i = 1; i < words.length; i++) {
            result.append("For want of a " + words[i - 1] + " the " + words[i] + " was lost.\n");
        }
        result.append("And all for the want of a " + words[0] + ".");
        return result.toString();
    }
}
